src/writer/_dirent.h

```diff
@@ -101,9 +151,9 @@ namespace zim
           { }
 
         char getNamespace() const                { return ns; }
-        const std::string& getTitle() const      { return title.empty() ? path : title; }
-        const std::string& getRealTitle() const  { return title; }
-        const std::string& getPath() const       { return path; }
+        std::string getTitle() const      { return title.empty() ? path : title; }
+        std::string getRealTitle() const  { return title; }
+        std::string getPath() const       { return path; }
 
```

Comment:
Why don't you return a const reference to `TinyString`?

Reply:
The `title` and `path` are stored in a `PathTitleTinyString` and so we must recreate a string each time.